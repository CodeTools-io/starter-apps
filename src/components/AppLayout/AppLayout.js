import React from 'react';
import { Box, Grid } from 'grommet';

export default function AppLayout({ children, ...props }) {
  return (
    <Box className="AppLayout" fill {...props}>
      <Grid
        rows={['xsmall', 'auto', 'auto']}
        columns={['1/4', '1/4', '1/4', '1/4']}
        areas={[
          ['sidebar-header', 'header', 'header', 'header'],
          ['sidebar', 'main', 'main', 'main'],
          ['sidebar-footer', 'footer', 'footer', 'footer'],
        ]}
        fill
      >
        {children}
      </Grid>
    </Box>
  );
}
