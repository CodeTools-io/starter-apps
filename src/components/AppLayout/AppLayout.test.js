import React from 'react';
import AppLayout from './AppLayout';

describe('AppLayout', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <AppLayout>
        <p>Content for the AppLayout</p>
      </AppLayout>
    );

    expect(container).toMatchSnapshot();
  });
});
