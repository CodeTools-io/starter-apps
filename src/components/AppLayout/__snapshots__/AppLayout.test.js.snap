// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 iNKrsm AppLayout"
  >
    <div
      class="StyledGrid-sc-1wofa1l-0 fBOymk"
    >
      <p>
        Content for the AppLayout
      </p>
    </div>
  </div>
</div>
`;
